Sources/Core/BidirectionalCollection.swift

```diff
@@ -0,0 +1,22 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+public protocol BidirectionalCollection: Collection
+  where SubSequence: BidirectionalCollection, Indices: BidirectionalCollection {
+  override var indices: Indices { get }
+
+  override subscript(_: Range<Index>) -> SubSequence { get }
+
+  func index(before i: Index) -> Index
+  func formIndex(before i: inout Index)
+
+  override func index(after i: Index) -> Index
+  override func formIndex(after i: inout Index)
```

Comment:
I think it is better to keep all the overridden functions near one another.

Reply:
Not 100% sure what kind of grouping you'd like to see. I've rearranged them so that the `index(after:)` comes after `index(before:)`

Reply:
Think of this way: you have 0 context.  You see that this protocol is a refinement of `Collection`.  What is `Collection`? 🤷‍♂️ 

However, if you first have all the overridden properties you can clearly identify which properties are being refined and which are extensions.